src/sysupdate/sysupdate-resource.c

```diff
@@ -132,16 +119,42 @@ static int resource_load_from_directory(
                         return log_error_errno(errno, "Failed to stat %s/%s: %m", rr->path, de->d_name);
                 }
 
-                if ((st.st_mode & S_IFMT) != m)
+                if (!(S_ISDIR(st.st_mode) && S_ISREG(m)) && ((st.st_mode & S_IFMT) != m))
                         continue;
 
-                r = pattern_match_many(rr->patterns, de->d_name, &extracted_fields);
-                if (r < 0)
+                if (relpath != NULL)
```

Comment:
as per coding style, for pointer null checks we rely on C's downgrade to bool feature, i.e. change to `if (relpath)`